### ISSUE TITLE
Thanks!

### ISSUE THREAD

**Issue by coderschoolreview** [haantran96/ML-Week-3#1]
The goal of this assignment was to introduce you to three new Natural Language Processing techniques, and to understand how to perform some basic sentiment analysis on song lyrics using these methods. You learned how to clean and prepare textual information for NLP, and then apply the following approaches: Bag Of Words, TF-IDF, and Doc2Vec. You used your prior knowledge of Python estimators, feature selection, and parameter optimization techniques to produce feature vectors from these NLP methods to make predictions on the moods of songs using their lyrics.

Good job overall! Nice work trying in trying the bonus story of plotting K Means Clusters. Also, you didn't really have to do all the `TaggedLineDocument` and `.txt` file creation stuff for `Doc2Vec` (the helper function given to you does all that), but very good job in exploring it in detail! 

Here are some main points:
* You mentioned you were confused why `SVC` gave you a really poor score in tf-idf (score of 0 for `sad`). This is because `SVC` is most sensitive to parameters. Unfortunately, it is hard to have an intuition about the right parameters for `SVC` (i.e. what value to set `C` and `gamma` to). The default values can sometimes be really bad -- so for `SVC` you almost *always* want to run a `GridSearchCV` to find out the best parameters in order to get good results. So next time try doing that, and you should see a much better score.

* A `test_size` of `0.5` is pretty high! It is hard to see good results with that. I would suggest not going beyond `0.3`.

* Your `shuffle` function actually does **not** randomize your data! This is because you assign `X` and `y` *before* you shuffle the data. You should first shuffle your data and *then* assign `X` and `y`. (Also, `df_lyrics` is not being re-assigned to itself, so you need to also make sure to do that before assigning `X` and `y`). Let me know if you have any questions about this.
A simpler way to ensure your data is randomized is to instead randomize the seed in your `train_test_split` call i.e. use random numbers for `random_state` each time. This is what it is there for :)

* In your 
`sources = {'test-celebratory.txt':'TEST_CELEBRATORY', 'test-sad.txt':'TEST_SAD', 'train-celebratory.txt':'TRAIN_CELEBRATORY', 'train-pos.txt':'TRAIN_SAD'}`
It seems you use `train-pos.txt` which does not exist, so perhaps your `Doc2Vec` model was not fully trained?

* Great job in trying to plot the cluster data for KMeans. I have not used TruncatedSVD so unfortunately I can't really comment on your results. However, the assumption you made about wanting to try and fit to 2 clusters because we have 2 moods, is correct.

Overall, good work!
